hydrator-common/src/main/java/co/cask/hydrator/common/LineageRecorder.java

```diff
@@ -43,18 +48,28 @@ public LineageRecorder(BatchContext context, String dataset) {
   }
 
   /**
-   * Creates an external dataset with given schema is not null
+   * Creates an external dataset if a dataset with the given {@link LineageRecorder#dataset} name does not already
+   * exists. If a non null schema is provided then the external dataset will have the given schema upon creation.
    *
    * @param schema the schema of the external dataset
+   * @throws RuntimeException if the dataset creation fails
    */
-  public void createDataset(@Nullable String schema) {
-    if (schema != null) {
-      try {
-        context.createDataset(dataset, Constants.EXTERNAL_DATASET_TYPE,
-                              DatasetProperties.of(Collections.singletonMap(DatasetProperties.SCHEMA, schema)));
-      } catch (DatasetManagementException e) {
-        Throwables.propagate(e);
+  public void createExternalDataset(@Nullable Schema schema) {
+    DatasetProperties datasetProperties;
+    if (schema == null) {
+      datasetProperties = DatasetProperties.EMPTY;
+    } else {
+      datasetProperties = DatasetProperties.of(Collections.singletonMap(DatasetProperties.SCHEMA, schema.toString()));
+    }
+    try {
+      if (!context.datasetExists(dataset)) {
+        context.createDataset(dataset, Constants.EXTERNAL_DATASET_TYPE, datasetProperties);
+      } else {
+        LOG.info("Failed to create dataset {} with schema {}. A dataset with the given name already exists.",
```

Comment:
this is not a failure and should not be logged as one. Don't think this needs to be logged at all, it will happen whenever a pipeline is run more than once.